WcaOnRails/app/webpacker/components/CompetitionOverview/useDelegatesData.js

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+import { useInfiniteQuery } from '@tanstack/react-query';
+
+import { delegatesApiUrl, WCA_API_PAGINATION } from '../../lib/requests/routes.js.erb';
+import { fetchJsonOrError } from '../../lib/requests/fetchWithAuthenticityToken';
+
+const useDelegatesData = () => {
+  const [delegatesData, setDelegatesData] = useState([]);
```

Comment:
Would it work to just define
```
const delegatesData = rawDelegatesData?.pages?.flatMap((page) => page.data)
```
(possibly with `useMemo`), or is the effect & state necessary?

Reply:
The state is not necessary! The effect needs to stay to make it keep fetching as long as new data is coming in (unless there's some other way to do a while loop here?). Also now I realize what you meant with the `flatMap` thing haha.

Reply:
This means it also isn't necessary to do `useState` for competitionData, hm.

Reply:
And now without the `useState` but with a `const` switching between tabs look noticably smoother too. Experiencing firsthand not to overuse states 💀